### PR TITLE
store: fix closest_replicas can't work during follower read

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

Problem Summary:

### What is changed and how it works?

fix closest_replicas can't work during follower read

### Check List 

Tests 

- [x] Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: distsql/request_builder.go [hunk 1]
```diff
@@ -58,12 +58,10 @@ func (builder *RequestBuilder) Build() (*kv.Request, error) {
 			},
 		}
 	}
-	failpoint.Inject("assertRequestBuilderStalenessOption", func(val failpoint.Value) {
+	failpoint.Inject("assertRequestBuilderReplicaOption", func(val failpoint.Value) {
 		assertScope := val.(string)
-		if len(assertScope) > 0 {
-			if builder.IsStaleness && assertScope != builder.ReadReplicaScope {
-				panic("request builder get staleness option fail")
-			}
```
Comment (Yisaer) [inline distsql/request_builder.go:66]: The reason why we didn't find bug by unit test for follower read is due to the asserting condition is made for stale read, thus we change the asserting condition in order to detect follower read's situation.

**Review by xhebox — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**